bwameth.py

```diff
@@ -311,8 +334,20 @@ def name(f):
 def bwa_mem(fa, fq_convert_cmd, extra_args, threads=1, rg=None,
             paired=True, set_as_failed=None, do_not_penalize_chimeras=False):
     conv_fa = convert_fasta(fa, just_name=True)
-    if not is_newer_b(conv_fa, (conv_fa + '.amb', conv_fa + '.sa')):
-        raise BWAMethException("first run bwameth.py index %s" % fa)
+
+    if is_newer_b(conv_fa, (conv_fa + '.amb', conv_fa + '.sa')):
+        idx = "mem1"
+        sys.stderr.write("--------------------\n")
+        sys.stderr.write("Found BWA MEM index\n")
+        
+    elif is_newer_b(conv_fa, (conv_fa + '.amb', conv_fa + '.pac')):
```

Comment:
it's been a while since I looked at this, but doesn't bwa mem generate a `.pac` file too?

Reply:
Yes it does. But mem2 index lacks the suffix array `.sa` which is what being checked for `mem1` index. 
We can also change the `.pac` for mem2 to `.2bit.64` 


```
├── mem1
│   ├── ref.fa
│   ├── ref.fa.bwameth.c2t
│   ├── ref.fa.bwameth.c2t.amb
│   ├── ref.fa.bwameth.c2t.ann
│   ├── ref.fa.bwameth.c2t.bwt
│   ├── ref.fa.bwameth.c2t.pac
│   └── ref.fa.bwameth.c2t.sa
├── mem2
│   ├── ref.fa
│   ├── ref.fa.bwameth.c2t
│   ├── ref.fa.bwameth.c2t.0123
│   ├── ref.fa.bwameth.c2t.amb
│   ├── ref.fa.bwameth.c2t.ann
│   ├── ref.fa.bwameth.c2t.bwt.2bit.64
│   └── ref.fa.bwameth.c2t.pac

```

Reply:
makes sense. thanks for the clarification